src/taipy/core/config/migration_config.py

```diff
@@ -0,0 +1,192 @@
+
+import collections.abc
+from copy import deepcopy
+from typing import Any, Callable, Dict, Optional, Union
+
+from taipy.config._config import _Config
+from taipy.config.common._template_handler import _TemplateHandler as _tpl
+from taipy.config.config import Config
+from taipy.config.unique_section import UniqueSection
+
+from .._version._version_manager import _VersionManager
+from ..common.alias import DataNodeId, PipelineId, ScenarioId, TaskId
+from .data_node_config import DataNodeConfig
+from .pipeline_config import PipelineConfig
+from .scenario_config import ScenarioConfig
+from .task_config import TaskConfig
+
+
+class MigrationConfig(UniqueSection):
+    """
+    Configuration fields needed to register migration functions from an old version to newer one.
+
+    Attributes:
+        migration_fcts (Dict[str, Dict[str, Callable]]): A dictionary that maps the version that entities are
+            migrated from to the migration functions.
+        **properties (dict[str, Any]): A dictionary of additional properties.
+    """
+
+    name = "VERSION_MIGRATION"
+
+    _MIGRATION_FCTS_KEY = "migration_fcts"
+
+    def __init__(
+        self,
+        migration_fcts: Dict[str, Dict[str, Callable]],
+        **properties,
+    ):
+        self.migration_fcts = migration_fcts
+
+        super().__init__(**properties)
+
+    def __copy__(self):
+        return MigrationConfig(
+            deepcopy(self.migration_fcts),
+            **deepcopy(self._properties),
+        )
+
+    def __getattr__(self, item: str) -> Optional[Any]:
+        return _tpl._replace_templates(self._properties.get(item))  # type: ignore
+
+    @classmethod
+    def default_config(cls):
+        return MigrationConfig({})
+
+    def _to_dict(self):
+        return {
+            self._MIGRATION_FCTS_KEY: self.migration_fcts,
+            **self._properties,
+        }
+
+    @classmethod
+    def _from_dict(cls, as_dict: Dict[str, Any], id: str, config: Optional[_Config]):
+        return MigrationConfig(**as_dict)
+
+    def _update(self, as_dict, default_section=None):
+        def deep_update(d, u):
+            for k, v in u.items():
+                if isinstance(v, collections.abc.Mapping):
+                    d[k] = deep_update(d.get(k, {}), v)
+                else:
+                    d[k] = v
+            return d
+
+        migration_fcts = as_dict.pop(self._MIGRATION_FCTS_KEY)
+        for source_version in list(migration_fcts.keys()):
+            if (proper_version := _VersionManager._replace_version_number(source_version)) != source_version:
+                migration_fcts[proper_version] = migration_fcts.pop(source_version)
+
+        deep_update(self.migration_fcts, migration_fcts)
+
+        self._properties.update(as_dict)
+
+    @staticmethod
+    def _add_data_node_migration_function(
+        source_version: str,
+        data_node_config: Union[DataNodeConfig, DataNodeId],
+        migration_fct: Callable,
+        **properties,
+    ):
```

Comment:
As a user, when developing version 2.0 of my app, I want to provide a single function to migrate old entities (from multiple old versions <2.0) to the single version 2.0.
So the API should ask for the target version, not the source version.

```suggestion
    def _add_data_node_migration_function(
        target_version: str,
        data_node_config: Union[DataNodeConfig, DataNodeId],
        migration_fct: Callable,
        **properties,
    ):
```

Ex:

Config._add_task_migration_function("2.0", my_datanode_cfg, my_fct)